src/RazorSdk/SourceGenerators/SourceGeneratorProjectItem.cs

```diff
@@ -39,10 +49,12 @@ public SourceGeneratorProjectItem(string basePath, string filePath, string relat
 
         public override string FileKind => _fileKind ?? base.FileKind;
 
-        public override string CssScope { get; }
+        public override string? CssScope { get; }
 
-        public override Stream Read()
-            => throw new NotSupportedException("This API should not be invoked. We should instead be relying on " +
-                "the RazorSourceDocument associated with this item instead.");
+        public override Stream? Read()
```

Comment:
I would rather we keep the exception. Getting here indicates a bug in our code.

Reply:
Sure. Brought it back.